src/Data/Cardinality.php

```diff
@@ -0,0 +1,100 @@
+<?php
+namespace LORIS\Data;
+
+/**
+ * Cardinality represents the number of data points which
+ * apply to the scope of a data type.
+ *
+ * Since the Cardinality class represents an enumeration, the
+ * class is final.
+ *
+ * @license http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ */
+final class Cardinality implements \JsonSerializable
+{
+    // Valid cardinality types for data to apply to.
+
+    /**
+     * A Unique Cardinality signifies that the data is unique
+     * across the scope. Examples of unique data are CandID
+     * for the candidate scope or VisitLabel for the Session
+     * scope.
+     */
+    const UNIQUE = 1;
+
+    /**
+     * A Single Cardinality signifies that each data point in
+     * the scope should have exactly one value. For instance,
+     * date of birth for a candidate in the candidate scope.
+     */
+    const SINGLE = 2;
+
+    /**
+     * An Optional Cardinality signifies that each data point
+     * in the scope may have zero or one value. For instance,
+     * the date of death for a candidate in the candidate scope.
+     */
+    const OPTIONAL = 3;
+
+    /**
+     * A Many Cardinality signifies that each data point will
+     * have zero or more values associated. For instance,
+     * the T1 scans acquired at a session.
```

Comment:
feels like the example is a bit too specific, why not use something like `For instance, site affiliations for a user` for us non-imaging people

Reply:
I don't mind changing the example but I think site/project affiliations is a bad example because there's no "user" scoped data, and the example might get confusing since site/project affiliations exist on both candidate and session scopes.

Reply:
how about using the candidate/session relationship then ? a candidate can have 0 to many sessions

Reply:
That would be confusing too though, because there's the session scope and there's no variable named "Session", and session based data is usually described as 1:1 with a session scope, not 1:many with the candidate.